worlds/tww/randomizers/ItemPool.py

```diff
@@ -110,6 +110,15 @@ def get_pool_core(world: "TWWWorld") -> tuple[list[str], list[str]]:
             else:
                 filler_pool.extend([item] * data.quantity)
 
+    # If the player starts with a sword, add one to the precollected items list and remove one from the item pool.
+    if world.options.sword_mode == "start_with_sword":
+        precollected_items.append("Progressive Sword")
+        progression_pool.remove("Progressive Sword")
+    # Or, if it's swordless mode, remove all swords from the item pool.
+    elif world.options.sword_mode == "swordless":
+        while "Progressive Sword" in useful_pool:
+            useful_pool.remove("Progressive Sword")
```

Comment:
You can just do this instead, it's faster:
```suggestion
        useful_pool = [item for item in useful_pool if item != "Progressive Sword"]
```